inc/dashboard/namespace.php

```diff
@@ -0,0 +1,456 @@
+			'label' => __( '30 days', 'altis-analytics' ),
+		],
+		[
+			'days' => 90,
+			'label' => __( '90 days', 'altis-analytics' ),
+		],
+	];
+
+	$days = get_days_view();
+	?>
+	<div class="xb-analytics-date-range">
+		<?php
+		foreach ( $ranges as $current ) {
+			// Compare the current days value with the current days value pulled from the url query variables.
+			if ( $days === $current['days'] ) {
+				$selected = 'selected';
+			} else {
+				$selected = '';
+			}
+
+			printf(
+				wp_kses_post( '<a href="%1$s" class="date-range-button %2$s">%3$s</a>' ),
+				esc_url_raw( add_query_arg( [ 'days' => $current['days'] ] ) ),
+				esc_html( $selected ),
+				esc_html( $current['label'] )
+			);
+		}
+		?>
+	</div>
+	<?php
+}
+
+/**
+ * Render the block name column data.
+ */
+function render_block_column() {
+	global $post;
+	?>
+	<strong><a href="<?php echo esc_url_raw( admin_url( '/admin.php?page=xb-analytics&post=' . $post->ID ) ); ?>"><?php echo esc_attr( $post->post_title ); ?></a></strong>
+	<?php
+}
+
+/**
+ * Render the views column data.
+ */
+function render_views() {
+	global $post;
+	$list = get_views_list( get_days_view() );
+	$views = $list[ $post->post_name ]['views'];
+	?>
+	<div class="post--views"><?php echo number_format_i18n( $views ); ?></div>
+	<?php
+}
+
+/**
+ * Render the average conversion rate column data.
+ */
+function render_average_conversion_rate() {
+	global $post;
+	$list = get_views_list( get_days_view() );
+	$rate = round( $list[ $post->post_name ]['avg_conversion_rate'] * 100 );
+	?>
+	<div class="post--avg-conversion-rate"><?php echo absint( $rate ); ?>%
+	<?php
+}
+
+/**
+ * Return the days url query string.
+ *
+ * @return int The number of days to show insights for. Defaults to 7.
+ */
+function get_days_view() : int {
+	return isset( $_GET['days'] ) ? wp_unslash( absint( $_GET['days'] ) ) : 7;
+}
+
+/**
+ * Calculate average conversion rate.
+ *
+ * @param int $conversions The number of conversions for a block.
+ * @param int $views The number of views for a block.
+ *
+ * @return float The conversion rate (calculated by conversions / views).
+ */
+function calculate_average_conversion_rate( int $conversions = 0, int $views = 0 ) : float {
+
+	// Avoid division by zero.
+	if ( $views === 0 ) {
+		return 0;
+	}
+
+	return $conversions / $views;
+}
+
+/**
+ * Aggregate and map the data from the ES query.
+ *
+ * @param array $buckets The ElasticSearch query data.
+ *
+ * @return array An array of aggregated data.
+ */
+function get_aggregate_data( array $buckets ) : array {
+	$data = [];
+
+	foreach ( $buckets as $block ) {
+		$block_id = $block['key'];
+		$views = $block['views']['uniques']['value'] ?? 0;
+		$conversions = $block['conversions']['uniques']['value'] ?? 0;
+		$conversion_rate = $block['conversionRate']['value'] ?? 0;
+
+		$data[ $block_id ]['views'] = $views;
+		$data[ $block_id ]['conversions'] = $conversions;
+		$data[ $block_id ]['avg_conversion_rate'] = $conversion_rate;
```

Comment:
Just occurred to me but the `avg_` bit isn't really what we're looking at here. We're not taking an average of conversion rates across a bunch of blocks or time ranges so this is just the conversion rate, no averages involved.

Reply:
Hmm...good point, so we'll need to change the column headers and microcopy as well.